src/app/campaign/flight/flight.component.ts

```diff
@@ -90,29 +90,34 @@ export class FlightComponent implements OnInit {
   }
 
   // emits updates when reactive form fields change
-  formStatusChanged(flight?: Flight) {
+  formStatusChanged(flight: Flight, changed?: boolean) {
     if (this.emitFormUpdates) {
       this.flightUpdate.emit({
         flight: { ...flight, totalGoal: this.flight.totalGoal },
-        changed: this.flightForm.dirty,
+        changed,
         valid: this.flightForm.valid
       });
     }
   }
 
   onDateRangeChange({ startAt, endAt }: { startAt?: Date; endAt?: Date }) {
-    if (this.emitFormUpdates) {
-      this.flightUpdate.emit({
-        flight: {
-          ...this.flight,
-          ...(startAt && { startAt }),
-          ...(endAt && { endAt }),
-          totalGoal: this.flight.totalGoal
-        },
-        changed: true,
-        valid: this.flightForm.valid
-      });
```

Comment:
I was having problems with invalid form state on these dates when they were empty for duplicated campaigns. I saw they weren't yet set at this point in the reactive form because these datapickers are not reactive form connected. I thought it should call the same formStatusChanged function. We have several tickets and some open PRs that I need to get back to at some point on the material date range pickers and reactive form start and end date validation.